### DISCUSSION TITLE
RGB Integration

### DISCUSSION THREAD

**Discussion by Quentinvanhumbeck** [gicamm/homeassistant-comelit#73] [Ideas]
Hi,

Could it be possible to integrate RGB lights?
I could try and test them, but not experienced with the coding. So if anyone else could do it quicker, please don't hesitate :-)

Thx Q

**Comment by gicamm** [comment]
Hi, this could be a great feature, you can start monitoring the MQTT messages so we can find the component and the identifier. So we could start implementing some things.
Do you have any idea how to connect to the Comelit HUB and look at the MQTT messages?
If not, you can look at this
[https://github.com/gicamm/homeassistant-comelit/blob/master/test/mqtt.txt](https://github.com/gicamm/homeassistant-comelit/blob/master/test/mqtt.txt)

**Reply by Quentinvanhumbeck** [reply]
Hi,

This is the MQTT message to put the light on 100% in "blue".

`{"req_type": 0,
  "req_sub_type": -1,
  "obj_id": "DOM#LT#80.5",
  "out_data": [
    {"id": "DOM#LT#80.5",
      "type": 3,
      "sub_type": 2,
      "sched_status": "0",
      "status": "7",
      "powerst": "7",
      "rgb_value": ["0", "44", "248"],
      "rgb_value_raw": ["0", "44", "248"],
      "rgb_value_888": ["0", "44", "255"],
      "bright": "255"} ] }`

I already succeeded in changing the brightness by adding the subtype to the existing code, but the rgb values are still not working.

subtype 6 is btw for DALI lights, could be added to the code as well

def update_light(self, id, description, data):
        try:
            _LOGGER.debug("update_light: %s has data %s", description, data)
            
            # Dimmable light (3=light, 4=dimmable)
            if data["type"] == 3 and data["sub_type"] == 4:
                brightness = int(data["bright"])
            **elif data["type"] == 3 and data["sub_type"] == 2:
                brightness = int(data["bright"])
            elif data["type"] == 3 and data["sub_type"] == 6:
                brightness = int(data["bright"])**
            else:
                brightness = None

            if data["status"] == "1":
                state = STATE_ON
            else:
                state = STATE_OFF

            light = ComelitLight(id, description, state, brightness, CommandHub(self))
            if id not in self.lights:  # Add the new sensor
                if hasattr(self, 'light_add_entities'):
                    self.light_add_entities([light])
                    self.lights[id] = light
                    _LOGGER.info("added the light %s %s", description, light.entity_name)
            else:
                _LOGGER.debug(f"updating the light {description} {light.entity_name} with state {state}")
                self.lights[id].update_state(state)
        except Exception as e:
            _LOGGER.exception("Error updating light %s", e)
